services/frontend-service/src/ui/ActionDetails.tsx

```diff
@@ -0,0 +1,197 @@
+/*This file is part of kuberpult.
+
+Kuberpult is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+Kuberpult is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with kuberpult.  If not, see <http://www.gnu.org/licenses/>.
+
+Copyright 2021 freiheit.com*/
+import { BatchAction } from '../api/api';
+import {
+    DeleteForeverRounded,
+    DeleteOutlineRounded,
+    Error,
+    LockOpenRounded,
+    LockRounded,
+    MoveToInboxRounded,
+} from '@material-ui/icons';
+import * as React from 'react';
+
+export enum ActionTypes {
+    Deploy,
+    PrepareUndeploy,
+    Undeploy,
+    CreateEnvironmentLock,
+    DeleteEnvironmentLock,
+    CreateApplicationLock,
+    DeleteApplicationLock,
+    UNKNOWN,
+}
+
+type ActionDetails = {
+    type: ActionTypes;
+    name: string;
+    summary: string;
+    dialogTitle: string;
+    notMessageSuccess: string;
+    notMessageFail: string;
+    description?: string;
+    icon: React.ReactElement;
+
+    // action details optional
+    environment?: string;
+    application?: string;
+    lockId?: string;
+    lockMessage?: string;
+    version?: number;
+};
+export const GetActionDetails = (action: BatchAction): ActionDetails => {
```

Comment:
Please use lowercase for functions. Uppercase is reserved for classes and react components.

Reply:
I changed the name in the last commit!